ui/app/pages/permissions-connect/permissions-connect.component.js

```diff
@@ -142,11 +135,56 @@ export default class PermissionConnect extends Component {
       return history.push(DEFAULT_ROUTE)
     }
 
-    if (getEnvironmentType() === ENVIRONMENT_TYPE_FULLSCREEN) {
+    if (getEnvironmentType() === ENVIRONMENT_TYPE_FULLSCREEN || getEnvironmentType() === ENVIRONMENT_TYPE_NOTIFICATION) {
       window.addEventListener('beforeunload', this.beforeUnload)
     }
   }
 
+  cancelPermissionsRequest = async (requestId) => {
+    const { rejectPermissionsRequest } = this.props
+    if (requestId) {
+      await rejectPermissionsRequest(requestId)
+
+      if (getEnvironmentType() === ENVIRONMENT_TYPE_FULLSCREEN) {
+        const currentTab = await global.platform.currentTab()
+        global.platform.closeTab(currentTab.id)
```

Comment:
Addressed in 829d2cbf3

Reply:
I'm not sure how this has been addressed; I was suggesting that `window.close` could be used instead of both `platform.closeCurrentWindow` and `platform.closeTab`, but now you're using `platform.closeCurrentWindow` even on fulscreen? Wouldn't this close the window even if it had other tabs? That seems very wrong.

Reply:
Okay, actually addressed in 619e8c117